python/ray/rllib/optimizers/multi_gpu_optimizer.py

```diff
@@ -116,13 +120,12 @@ def step(self, postprocess_fn=None):
                         self.sess,
                         permutation[batch_index] * self.per_device_batch_size)
                     for k, v in batch_fetches.items():
-                        iter_extra_fetches[k] += [v]
-                for k, v in iter_extra_fetches.items():
-                    all_extra_fetches[k] += [v]
+                        iter_extra_fetches[k].append(v)
+                print(i, _averaged(iter_extra_fetches))
```

Comment:
This was incorrect before; you want to return the last epoch's stats, not the mean across all epochs.

Reply:
Eh, I'm not sure why returning all epoch stats makes sense in any scenario, unless your epochs were very small and the values noisy.